src/pages/waypoint/docs/index.tsx

```diff
@@ -0,0 +1,172 @@
+// Third-party imports
+import { MDXRemoteSerializeResult } from 'next-mdx-remote'
+
+// HashiCorp imports
+import { IconBox16 } from '@hashicorp/flight-icons/svg-react/box-16'
+import { IconTerminalScreen16 } from '@hashicorp/flight-icons/svg-react/terminal-screen-16'
+
+// Global imports
+import waypointData from 'data/waypoint.json'
+import { ProductData } from 'types/products'
+import SidebarSidecarLayout from 'layouts/sidebar-sidecar'
+import { getStaticGenerationFunctions } from 'layouts/sidebar-sidecar/server'
+import IconTileLogo from 'components/icon-tile-logo'
+import Heading from 'components/heading'
+import Text from 'components/text'
+import CardLink from 'components/card-link'
+import ProductDocsSearch from 'views/docs-view/components/product-docs-search'
+import DocsView from 'views/docs-view'
+import CardsGridList from 'components/cards-grid-list'
+import TruncateMaxLines from 'components/truncate-max-lines'
+import devDotStyles from 'components/dev-dot-content/dev-dot-content.module.css'
+
+// Local imports
+import pageContent from './content.json'
+import s from './style.module.css'
+
+// Constants
+const basePath = 'docs'
+const baseName = 'Docs'
+const product = waypointData as ProductData
+const iconMap = {
+  box: <IconBox16 />,
+  'terminal-screen': <IconTerminalScreen16 />,
+}
+const heroCards = [
+  {
+    iconName: 'terminal-screen',
+    link: '/waypoint/commands',
+    text: 'CLI',
+  },
+  {
+    iconName: 'box',
+    link: '/waypoint/plugins',
+    text: 'Plugins',
+  },
+]
+
+interface WaypointDocsLandingProps {
+  mdxSource: MDXRemoteSerializeResult
+}
+
+// TODO: abstract the markup to a new view, such as RootDocsPathLandingView
+const WaypointDocsLanding = ({ mdxSource }: WaypointDocsLandingProps) => {
+  const { pageSubtitle } = pageContent
+
+  return (
+    <>
+      {__config.flags.enable_product_docs_search ? <ProductDocsSearch /> : null}
+      <div className={s.pageHeroWrapper}>
+        <div className={s.pageHeroText}>
+          <IconTileLogo productSlug="waypoint" />
+          <div>
+            <Heading className={s.pageTitle} level={1} size={500} weight="bold">
+              Waypoint Documentation
+            </Heading>
+            <Text className={s.pageSubtitle}>{pageSubtitle}</Text>
+          </div>
+        </div>
+        <ul className={s.heroCardList}>
```

Comment:
There's [something similar in `product-landing`](https://github.com/hashicorp/dev-portal/blob/2428520b75ce77861a96d906d2329409b0364c4b/src/views/product-landing/components/icon-cards/index.tsx#L10) that uses `components/cards-grid-list`. Might be ripe for elevation & re-use here! (I probably should have lifted the `LinkedIconCard` into `components` initially 😅 )

Reply:
Sweet!! I'll definitely leverage these things and I'll lift them to the higher folder as well. 😊